app/networkapi/filebrowser_s3/templatetags/s3thumbnails.py

```diff
@@ -0,0 +1,186 @@
+import os
+
+try:
+    from urllib.parse import quote, unquote
+except ImportError:
+    from urllib import quote, unquote
+
+from django.core.files import File
+from django.core.files.storage import default_storage
+from mezzanine.conf import settings
+from mezzanine import template
+
+register = template.Library()
+
+
+@register.simple_tag(name='thumbnail')
+def thumbnail(image_url, width, height, upscale=True, quality=95, left=.5,
+              top=.5, padding=False, padding_color="#fff"):
+    """
+    Given the URL to an image, resizes the image using the given width
+    and height on the first time it is requested, and returns the URL
+    to the new resized image. If width or height are zero then original
+    ratio is maintained. When ``upscale`` is False, images smaller than
+    the given size will not be grown to fill that size. The given width
+    and height thus act as maximum dimensions.
+    """
+    if not image_url:
+        return ""
+    try:
+        from PIL import Image, ImageFile, ImageOps
+    except ImportError:
+        return ""
+
+    image_url = unquote(str(image_url)).split("?")[0]
+    if image_url.startswith(settings.MEDIA_URL):
+        image_url = image_url.replace(settings.MEDIA_URL, "", 1)
+    image_dir, image_name = os.path.split(image_url)
+    image_prefix, image_ext = os.path.splitext(image_name)
+    filetype = {".png": "PNG", ".gif": "GIF"}.get(image_ext, "JPEG")
+    thumb_name = "%s-%sx%s" % (image_prefix, width, height)
+    if not upscale:
+        thumb_name += "-no-upscale"
+    if left != .5 or top != .5:
+        left = min(1, max(0, left))
+        top = min(1, max(0, top))
+        thumb_name = "%s-%sx%s" % (thumb_name, left, top)
+    thumb_name += "-padded-%s" % padding_color if padding else ""
+    thumb_name = "%s%s" % (thumb_name, image_ext)
+
+    # `image_name` is used here for the directory path, as each image
+    # requires its own sub-directory using its own name - this is so
+    # we can consistently delete all thumbnails for an individual
+    # image, which is something we do in filebrowser when a new image
+    # is written, allowing us to purge any previously generated
+    # thumbnails that may match a new image name.
+    thumb_dir = os.path.join(settings.MEDIA_ROOT, image_dir,
+                             settings.THUMBNAILS_DIR_NAME, image_name)
+    if not os.path.exists(thumb_dir):
+        try:
+            os.makedirs(thumb_dir)
+        except OSError:
+            pass
+
+    thumb_path = os.path.join(thumb_dir, thumb_name)
+    thumb_url = "%s/%s/%s" % (settings.THUMBNAILS_DIR_NAME,
+                              quote(image_name.encode("utf-8")),
+                              quote(thumb_name.encode("utf-8")))
+    image_url_path = os.path.dirname(image_url)
+    if image_url_path:
+        thumb_url = "%s/%s" % (image_url_path, thumb_url)
+
+    remote_url = "%s%s/%s" % (
```

Comment:
This template tag used to return just the relative url to the file, I've added `remote_url` and set it to be returned throughout the file.